airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java

```diff
@@ -168,6 +187,10 @@ ImmutablePair<Boolean, String> isSizeTriggered(final StreamDescriptor stream, fi
         AirbyteFileUtils.byteCountToDisplaySize(sizeOfRunningWorkersEstimate),
         AirbyteFileUtils.byteCountToDisplaySize(queueSizeAfterRunningWorkers));
 
+    if (isSizeTriggered) {
+      lastTimeCalledPerStream.put(stream, nowProvider.getAsLong());
+    }
+
```

Comment:
Can we just consolidate this logic in the `getNextStreamToFlush` method body? I don't think it needs to be split between the callsites

Reply:
I needed it for the U test, that's why I didn't put the logic in `getNextStreamToFlush`. The test is `TimeTriggerTest`